packages/core/strapi/src/cli/commands/generate-schema.ts

```diff
@@ -0,0 +1,210 @@
+import { createCommand } from 'commander';
+import { createStrapi, compileStrapi } from '@strapi/core';
+import type { Struct } from '@strapi/types';
+
+import * as fs from 'fs';
+import tsUtils from '@strapi/typescript-utils';
+import prettier from 'prettier';
+import type { StrapiCommand } from '../types';
+
+interface CmdOptions {
+  silent?: boolean;
+  outDir?: string;
+}
+
+// Filter to only include content types where the UID begins with "api::"
+// TODO: temporary restriction to API content types
+const restrictToApiContentTypes = (uid: string): boolean => {
+  try {
+    return uid.startsWith('api::');
+  } catch (e) {
+    console.error('Error checking UID:', e);
+    return false;
+  }
+};
+
+const allCrudOperationNames = ['create', 'update', 'delete', 'findOne', 'find'];
+
+const crudOperationNames = {
+  collectionType: allCrudOperationNames,
+  singleType: allCrudOperationNames.filter((op) => op !== 'find'),
+};
+
+const action = async ({ silent, outDir }: CmdOptions) => {
+  if (silent) {
+    console.log = () => {};
+  }
+
+  console.log('Generating schema.ts...');
+  const outputDir = outDir ?? 'types';
+
+  // Create and register a Strapi instance
+  const appContext = await compileStrapi();
+  const app = await createStrapi(appContext).register();
+
+  // Generate types with modifications. The Modifications are:
+  // - Filter out content types that are not API content types
+  // - Filter out attributes that are private or password
+  await tsUtils.generators.generate({
+    strapi: app,
+    pwd: appContext.appDir,
+    rootDir: outputDir,
+    logger: {
+      silent,
+    },
+    artifacts: {
+      contentTypes: true,
+      components: true,
+    },
+    filters: {
+      contentTypes(contentType: Struct.CollectionTypeSchema) {
+        return restrictToApiContentTypes(contentType.uid);
+      },
+    },
+    transforms: {
+      contentTypes(contentType: Struct.CollectionTypeSchema) {
+        // Strip private attributes
+        const attributes = Object.fromEntries(
+          Object.entries(contentType.attributes).filter(
+            ([, attr]) =>
+              attr.type !== 'password' &&
+              // @ts-expect-error TODO: fix this
+              !attr.private
+          )
+        );
+        return { ...contentType, attributes };
```

Comment:
To align the behavior with other APIs, we could use the content types utils (@strapi/utils) to whether an attribute is private or not

```suggestion
        const attributes = Object.fromEntries(
          Object.entries(contentType.attributes)
            // Skip passwords
            .filter(([, attribute]) => attribute.type !== 'password')
            // Strip private attributes
            .filter(([name]) => !contentTypesUtils.isPrivateAttribute(contentType, name))
        );

        return { ...contentType, attributes };
```